### PR TITLE
Add Bert python

### PR DESCRIPTION
**Pull request by dding3**
1. Add bert python version
2. Code refactor Transformer
3. Update Transformer example based on new interface
4. minor fix

### CODE REVIEW

**Comment by dding3** [comment]
jekins pass. http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/2142/

File: pyzoo/zoo/pipeline/api/keras/layers/self_attention.py [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+
+    def merge_heads(self, x):
+        p = auto.contiguous(Permute((2, 1, 3))(x))
+        sizes = p.get_output_shape()[1:]
+        merge_sizes = list(sizes[:-2] + (sizes[-1] * sizes[-2],))
+        m = Reshape(merge_sizes)(p)
+        return m
+
+    @classmethod
+    def init_with_default_embedding(cls, vocab=40990, seq_len=77, n_block=12, hidden_drop=0.1,
```
Comment (jason-dai) [inline pyzoo/zoo/pipeline/api/keras/layers/self_attention.py]: shall we change it to `init()`?

File: pyzoo/zoo/pipeline/api/keras/layers/self_attention.py [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+        return (- inputs[-1] + 1.0) * -10000.0, inputs[:-1], inputs
+
+    def gelu(self, x):
+        y = x / math.sqrt(2.0)
+        e = auto.erf(y)
+        y = x * 0.5 * (e + 1.0)
+        return y
+
+    @classmethod
+    def init_with_default_embedding(cls, vocab=40990, hidden_size=768, n_block=12, n_head=12,
```
Comment (jason-dai) [inline pyzoo/zoo/pipeline/api/keras/layers/self_attention.py]: shall we change it to `init()`?

**Comment by dding3** [comment]
Jekins pass. http://10.239.47.211:18888/job/ZOO-PR-Validation/2187/

File: pyzoo/zoo/pipeline/api/keras/layers/self_attention.py (hunk 2 above)
Comment (dding3) [reply pyzoo/zoo/pipeline/api/keras/layers/self_attention.py]: Update

File: pyzoo/zoo/pipeline/api/keras/layers/self_attention.py (hunk 1 above)
Comment (dding3) [reply pyzoo/zoo/pipeline/api/keras/layers/self_attention.py]: Update

**Pull request merged**